src/utlist.h

```diff
@@ -775,6 +842,40 @@ do {
  (head) = (add);                                                                               \
 } while (0)
 
+#define CDL_INSERT_INORDER(head,add,cmp)                                                       \
+  CDL_INSERT_INORDER2(head,add,cmp,prev,next)
+
+#define CDL_INSERT_INORDER2(head,add,cmp,prev,next)                                                       \
+  do {                                                                                         \
+    LDECLTYPE(head) _pos;                                                                      \
+    _CASTASGN(_pos, head);                                                                     \
+    if (head) {                                                                                \
+      CDL_LOWER_BOUND(_pos,add,cmp);                                                           \
+      CDL_APPEND_ELEM(head,_pos,add);                                                          \
+    } else {                                                                                   \
+      (head) = add;                                                                            \
+      (head)->next = (head);                                                                   \
+      (head)->prev = (head);                                                                   \
+    }                                                                                          \
+  } while (0)
+
+#define CDL_LOWER_BOUND(head,node,cmp)                                                         \
+  CDL_LOWER_BOUND2(head,node,cmp,prev,next)                                                    \
+
+#define CDL_LOWER_BOUND2(head,node,cmp,prev,next)                                              \
+  do {                                                                                         \
+    LDECLTYPE(head) _tmp;                                                                      \
+    _CASTASGN(_tmp, (head));                                                                   \
+    if (cmp(head, node) > 0) {                                                             \
+      (head) = NULL;                                                                           \
+    } else {                                                                                   \
+      for (; (head)->next != _tmp; head = (head)->next ) {                                     \
+        if (cmp((head)->next, node) > 0) {                                                   \
+          break;                                                                               \
+        }                                                                                      \
+      }                                                                                        \
+    }                                                                                          \
+  } while(0)
```

Comment:
Nit: `} while(0)` should be `} while (0)`. (And as noted above, it should start in column 1, not 3; and please fix the misaligned `\`s throughout.)